src/timetable/types.ts

```diff
@@ -0,0 +1,75 @@
+import { PopoverV2TriggerType } from "../popover-v2";
+import { ResourceScheme } from "../theme";
+
+export const ROW_BAR_COLOR_SEQUENCE = [
+    "#FFE6BB",
+    "#D8EFEB",
+    "#E6EAFE",
+    "#FAE4E5",
+    "#D3EEFC",
+] as const; // Assert to be a readonly tuple
+export const ROW_CELL_GAP = 2;
+export const ROW_INTERVAL = 15;
+export type RowBarColors = (typeof ROW_BAR_COLOR_SEQUENCE)[number];
+export type CellType = "filled" | "blocked" | "default";
+
+interface EmptyContentProps {
+    description: string;
+    illustrationScheme: ResourceScheme;
+}
+
+export interface RowData {
+    id: string;
+    name: string;
+    rowCells: RowCellData[];
+    subtitle?: string | JSX.Element | undefined;
+    rowHeaderCustomPopover?: CustomPopoverProps;
+    // HH:mm format, defaults to minTime
+    rowMinTime?: string;
+    // HH:mm format, defaults to maxTime
+    rowMaxTime?: string;
+}
+
+export interface CustomPopoverProps {
+    trigger: PopoverV2TriggerType;
+    content: string | JSX.Element;
+    width?: string | undefined;
+    padding?: string | undefined;
```

Comment:
what's the use case for custom padding?

Reply:
this was not explicitly mentioned in the Figma, but for the intended facilities use case, the filled slots popover needs to have a fixed width of 400px, and required some padding also

https://www.figma.com/design/sjlRZcuzHyFefY9uLjxZdJ/%5BBSG2.0%5D-Facilities-%E2%80%94-General-users?node-id=8920-143145&node-type=frame&t=VQFzI7xc2VfLzBlM-0

alternatively, it could be done up in such a way so that this component just renders the content, instead of wrapping it over the PopoverCard component?

Reply:
how about directly reusing the PopoverTrigger's `popoverContent`?

if `string | JSX.Element`, it renders PopoverV2 (which comes with mobile handling)

if `(() => React.ReactNode)`, it renders the custom element (which in your use case is a card of fixed width 400px)